packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -0,0 +1,427 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import 'terra-base/lib/baseStyles';
+import Field from 'terra-form/lib/Field';
+import Fieldset from 'terra-form/lib/Fieldset';
+import NumberField from 'terra-form/lib/NumberField';
+import DatePicker from 'terra-date-picker/lib/DatePicker';
+import SelectField from 'terra-form/lib/SelectField';
+
+
+const propTypes = {
+  /**
+   * The ISO 8601 string representation of the birth date to calculate an onset date from.
+   */
+  birthdate: PropTypes.string.isRequired,
+
+  /**
+   * The granularity of the onset date. YEAR, MONTH, and DATE are accepted.
+   */
+  granularity: PropTypes.string,
+
+  /**
+   * Name of the granularity select. The name should be unique.
+   */
+  granularitySelectName: PropTypes.string.isRequired,
+
+  /**
+   * A callback function to execute when a granularity is selected.
+   * The first parameter is the event. The second parameter is the changed granularity value.
+   */
+  granularitySelectOnChange: PropTypes.func,
+
+  /**
+   * The precision of the onset date. ABOUT, BEFORE, AFTER, and UNKNOWN are accepted.
+   */
+  precision: PropTypes.string,
+
+  /**
+   * Name of the precision select. The name should be unique.
+   */
+  precisionSelectName: PropTypes.string.isRequired,
+
+  /**
+   * A callback function to execute when a precision is selected.
+   * The first parameter is the event. The second parameter is the changed precision value.
+   */
+  precisionSelectOnChange: PropTypes.func,
+
+  /**
+   * The ISO 8601 string representation of the onset date to view/modify. Defaults to current date.
+   */
+  onsetDate: PropTypes.string,
+
+  /**
+   * Name of the onsetDate input. The name should be unique.
+   */
+  onsetDateInputName: PropTypes.string.isRequired,
+
+  /**
+   * A callback function to execute when a onsetDate is changed.
+   * The first parameter is the event. The second parameter is the changed onsetDate value.
+   */
+  onsetDateInputOnChange: PropTypes.func,
+};
+
+const defaultProps = {
+  birthdate: undefined,
+  granularity: 'MONTH',
+  granularitySelectName: undefined,
+  granularitySelectOnChange: undefined,
+  precision: 'ABOUT',
+  precisionSelectName: undefined,
+  precisionSelectOnChange: undefined,
+  onsetDate: moment().format('YYYY-MM-DD'),
+  onsetDateInputName: undefined,
+  onsetDateInputOnChange: undefined,
+};
+
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+};
+
+class OnsetPicker extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      granularity: this.props.granularity,
+      precision: this.props.precision,
+      onsetDate: this.props.onsetDate,
+      ageCalcCount: 1,
+      ageCalcDuration: 'weeks',
```

Comment:
Can we change these names? Perhaps age and ageUnit or if someone else has other ideas.

Reply:
Yeah rough names, changed to age and ageUnit.